pkg/analysis/importalias/analyzer.go

```diff
@@ -0,0 +1,179 @@
+					SuggestedFixes: []analysis.SuggestedFix{
+						{
+							Message: fmt.Sprintf("should replace %q with %q", alias, applicableAlias),
+							TextEdits: []analysis.TextEdit{
+								{
+									Pos:     importStmt.Pos(),
+									End:     importStmt.Name.End(),
+									NewText: []byte(applicableAlias),
+								},
+							},
+						},
+					},
+				},
+			)
+			return
+		}
+		if ok, lintErrMsg := checkAliasName(aliasSlice, pathSlice, pass); !ok {
+			applicableAlias := getAliasFix(pathSlice)
+			pass.Report(
+				analysis.Diagnostic{
+					Pos:     node.Pos(),
+					Message: fmt.Sprintf(lintErrMsg+" path: %s alias: %s", path, alias),
+					SuggestedFixes: []analysis.SuggestedFix{
+						{
+							Message: fmt.Sprintf("should replace %q with %q", alias, applicableAlias),
+							TextEdits: []analysis.TextEdit{
+								{
+									Pos:     importStmt.Pos(),
+									End:     importStmt.Name.End(),
+									NewText: []byte(applicableAlias),
+								},
+							},
+						},
+					},
+				},
+			)
+			return
+		}
+	})
+	return nil, nil
+}
+
+// checkVersion checks that if package name starts with `v` it's included in alias name
+func checkVersion(aliasLastWord string, pathSlice []string) bool {
+	versionExists, versionPos := packageVersion(pathSlice)
+	if !versionExists {
+		return true
+	}
+	return aliasLastWord == pathSlice[versionPos]
+
+}
+
+// checkAliasName check consistency in alias name
+func checkAliasName(aliasSlice []string, pathSlice []string, pass *analysis.Pass) (bool, string) {
+	lastUsedWordIndex := -1
+	for _, name := range aliasSlice {
+		// we don't check version rule here
+		if strings.HasPrefix(name, "v") || name == "" {
+			continue
+		}
+		usedWordIndex := searchString(pathSlice, name)
+
+		if usedWordIndex == len(pathSlice) {
+			return false, "used words in alias most be present in path"
+		}
+
+		if usedWordIndex <= lastUsedWordIndex {
+			return false, "order of words in alias should match words in path"
+		}
+
+		lastUsedWordIndex = usedWordIndex
+	}
+
+	if lastUsedWordIndex == -1 {
+		return false, "at least one word from path must be present in alias"
+	}
+
+	return true, ""
+}
+
+func getAliasFix(pathSlice []string) string {
+	versionExists, versionPos := packageVersion(pathSlice)
+	if !versionExists {
+		return pathSlice[len(pathSlice)-1]
+	}
+	if versionPos == len(pathSlice)-1 {
+		applicableAlias := pathSlice[len(pathSlice)-2] + "_" + pathSlice[versionPos]
+		return applicableAlias
+	}
+
+	applicableAlias := pathSlice[len(pathSlice)-1] + "_" + pathSlice[versionPos]
+	return applicableAlias
+}
+
+// packageVersion returns if some version specification exists in import path and it's position
+func packageVersion(pathSlice []string) (bool, int) {
+	for pos, value := range pathSlice {
+		if strings.HasPrefix(value, "v") {
+			return true, pos
+		}
+	}
+	return false, 0
+}
+
+func searchString(slice []string, word string) int {
+	for pos, value := range slice {
+		r, _ := regexp.Compile(word + "(s)?")
```

Comment:
Can you comment this? I think this would match "foo" and "foos"; is that what you are doing?

Reply:
yes I'm doing it on purpose, there are some imports in contour project using the word `apis` instead of `api`.
`contour_api_v1alpha1 "github.com/projectcontour/contour/apis/projectcontour/v1alpha1`
I got this example from contour `CONTRIBUTING.md` file